make all GogoLoco parameters Global

CobaltSpace opened the pull request:
This preserves the GogoLoco saved settings between uploads that change the amount of VRCFury components there are.

SenkyDragon commented:
Does this actually work? I thought vrchat saved parameters based on the position in the upload file, just like they are synchronized.

Toys0125 commented:
Putting all GoGoLoco parameters at the top of the Parameters list would be the best to prevent it from moving and work across FallBacks if set up the same way.

CobaltSpace commented:
Avatar save data is saved in a file that is very similar to json. It uses the names to load and save parameters. I moved my saved parameters around and added a saved dummy parameter to shift the VRCFury ones, and the avatars still loaded the saved settings.

For synchronization across fall backs, using global parameters would also enable placing the parameters in our avatar's actual parameters so the position is also consistent.

CobaltSpace commented:
An alternative idea would be to let components specify their own prefix instead of using a number as an option, and give this one something like `VFGo`

SenkyDragon commented:
Apologies for not handling this earlier. I've always wanted to rename gogo's VRCEmote to Go/VRCEmote (by request of another user), but never got around to adding rewrite support to Full Controller. I've hacked in this support in https://github.com/VRCFury/VRCFury/commit/cacfb2705992305e76073f02699386d028ab46fe , and we can clean it up some day ™️ if param rewrite support ever comes.
